pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -266,76 +263,66 @@ def _should_accumulate(self) -> bool:
         is_final_batch = self._num_training_batches_reached()
         return not (accumulation_done or is_final_batch)
 
-    def _track_epoch_end_reduce_metrics(
-        self, epoch_output: List[List[STEP_OUTPUT]], batch_end_outputs: STEP_OUTPUT
-    ) -> None:
-        """Adds the batch outputs to the epoch outputs and prepares reduction."""
-        hook_overridden = is_overridden("training_epoch_end", self.trainer.lightning_module)
-        if not hook_overridden:
-            return
+    @staticmethod
+    def _swap_optimizers_and_tbptt(tbptt_outputs: _BATCH_OUTPUTS_TYPE, n_opt: int) -> List[List[Dict[str, Any]]]:
```

Comment:
Do we want to move all tbptt utilties to their own folder, same for Lighting hooks ?

Reply:
> Do we want to move all tbptt utilties to their own folder

No because this case is a particularity of the `training_epoch_end(outputs)` format, not of `tbptt` itself. For that reason, this PR keeps this processing logic encapsulated in these `staticmethods` which ideally would be removed in the future.